test/dynamo/test_misc.py

```diff
@@ -11333,6 +11333,40 @@ def fn(x):
         self.assertEqual(r.y, torch.ones(2, 2) + 1)
         self.assertEqual(cnts.frame_count, 1)
 
+    def test_getattrvariable_as_python_constant(self):
+        from torch._dynamo.variables.misc import GetAttrVariable
+
+        @torch.compile(backend="eager")
+        def fn(x, rand1):
+            random.Random().setstate(rand1.getstate())
+            return x + rand1.random()
+
+        def get_rng(x):
+            rand1 = random.Random(1)
+            orig_random = rand1.random
+            rand1.random = lambda: orig_random()
+            return rand1
+
+        as_python_constant = GetAttrVariable.as_python_constant
```

Comment:
Can we use something like `unittest.mock` to patch and detect calls instead of manually patching?

Reply:
@williamwen42 

Yes, of course, silly me!

It turns out that `unitest.mock.patch.object` fits the job very nicely, take a look.